src/OpenTelemetry.Instrumentation.AspNet/TracerProviderBuilderExtensions.cs

```diff
@@ -42,7 +42,9 @@ public static class TracerProviderBuilderExtensions
             var aspnetOptions = new AspNetInstrumentationOptions();
             configureAspNetInstrumentationOptions?.Invoke(aspnetOptions);
 
-            builder.AddDiagnosticSourceInstrumentation((activitySource) => new AspNetInstrumentation(activitySource, aspnetOptions));
+            builder.AddInstrumentation(() => new AspNetInstrumentation(aspnetOptions));
+            builder.AddSource(typeof(AspNetInstrumentation).Assembly.GetName().Name);
```

Comment:
Why is this needed?

Reply:
We need to add the source as the instrumented activity's source will be updated from empty string to this in the OnStartActivityMethod. The ActivityStopped callback method checks if the activity's ActivitySource is present in the list of sources before calling Processor.OnEnd

Reply:
> The ActivityStopped callback method checks if the activity's ActivitySource is present in the list of sources before calling Processor.OnEnd

I dont see that check. ActivityStopped is simply checking if ActivitySource is empty or not.

Reply:
Ahh yes! I should update the ActivityStopped callback to check if the ActivitySource was added to the provider. This would require us to have a private field for Sources which will have all the sources that were added to the provider using AddSource. Does that sound good?

Reply:
The check for ActivitySource happens on `listener.ShouldListenTo`. If the source is not added then the ActivityStop callback will not be called `listener.ShouldListenTo` will return false for such an activity.